components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/util/APIFileUtils.java

```diff
@@ -332,19 +331,14 @@ public static void exportThumbnailToFileSystem(InputStream thumbnailInputStream,
      * @throws APIMgtDAOException if an error occurs while writing to the file
      */
     public static void writeStreamToFile(String path, InputStream inputStream) throws APIMgtDAOException {
-        FileOutputStream outputStream = null;
-        try {
-            outputStream = new FileOutputStream(path);
+        try (FileOutputStream outputStream = new FileOutputStream(path)) {
             IOUtils.copy(inputStream, outputStream);
-        } catch (FileNotFoundException e) {
-            throw new APIMgtDAOException(e);
-
         } catch (IOException e) {
-            throw new APIMgtDAOException("Unable to write to file at path: " + path, e);
-
+            String errorMsg = "Unable to write to file at path: " + path;
+            log.error(errorMsg, e);
+            throw new APIMgtDAOException(errorMsg, e);
         } finally {
             IOUtils.closeQuietly(inputStream);
```

Comment:
Ideally InputStream should not be closed by this function. Since the InputStream is created outside and passed as an argument to this function it is the responsibility of the code that constructs the InputStream to close it. If the code constructs the InputStream uses a 'try with resources' block then this will be taken care of